### PR TITLE
fix(oauth-provider): support prompt=none per OIDC spec

### LINKED ISSUE #7700: oauth-provider add support for `prompt=none`
### Is this suited for github?

- [x] Yes, this is suited for github

### Is your feature request related to a problem? Please describe.

PR #7665 added the enum for `prompt="none"` into the parameter list however the prompt is not actually supported internally.

### Describe the solution you'd like

Add support for `prompt="none"` for the oauth-provider plugin.

### Describe alternatives you've considered

na

### Additional context

Currently, there is no advertisement of the "none" prompt in metadata field `prompt_values_supported` so it's a feature request (not an issue).

### PR DESCRIPTION
**Pull request by himself65**
## Summary

Adds `prompt=none` support to the new `@better-auth/oauth-provider` plugin per the OIDC specification.

- When `prompt=none` is set and the user has **no active session**, redirect to `redirect_uri` with `error=login_required` instead of the login page
- When `prompt=none` is set and **consent is needed**, redirect to `redirect_uri` with `error=consent_required` instead of the consent page
- Add `"none"` to `prompt_values_supported` in the OIDC discovery metadata

## Context

PR #8398 (commit 9dff8c5) fixed `redirect_uri` validation for `prompt=none` in the **old** `oidc-provider` plugin (`packages/better-auth/src/plugins/oidc-provider/`), but the same feature was missing in the **new** `@better-auth/oauth-provider` package (`packages/oauth-provider/`). As noted in [this comment](https://github.com/better-auth/better-auth/issues/7700#issuecomment-4035862756), the fix was applied to the wrong plugin.

Note: The `redirect_uri` validation security concern from #8398 is already handled by the new plugin's existing flow — `redirect_uri` is validated against the registered client URIs before reaching the `prompt=none` check.

Closes #7700

## Test plan

- [x] `login_required` error returned when user is not authenticated with `prompt=none`
- [x] `consent_required` error returned when consent is needed with `prompt=none`
- [x] Invalid `redirect_uri` is rejected (not used for error redirects)
- [x] Metadata advertises `"none"` in `prompt_values_supported`
- [x] All existing oauth-provider tests still pass (50/50)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/oauth-provider/src/authorize.ts [hunk 1]
```diff
@@ -278,6 +278,19 @@ export async function authorizeEndpoint(
 	// Check for session
 	const session = await getSessionFromCtx(ctx);
 	if (!session || promptSet?.has("login") || promptSet?.has("create")) {
+		// Handle prompt=none per OIDC spec - must return error instead of redirecting to login
+		if (promptSet?.has("none")) {
```
[the remark this answers is not in the document]
Comment (dvanmali) [reply packages/oauth-provider/src/authorize.ts:282]: Currently restricted via `oauth.ts` query options. However, the location of this suggests both "login none" and "create none" are valid.

**Comment by dvanmali** [comment]
@himself65 I fixed some edge cases and added more tests in #8554
